### PR TITLE
state sync not asking itself

### LINKED ISSUE #1824: Unexpected error msgs in state_sync
`cargo test --color=always --all-features --package near --test sync_state_nodes sync_empty_state -- --exact --nocapture`

Test passes but with state sync error messages:
```
[2019-12-05T11:35:07Z ERROR near_client::sync] State sync: state download for shard 0 timed out in 10 seconds
[2019-12-05T11:35:07Z ERROR near_client::sync] State sync: state download for shard 1 timed out in 10 seconds
[2019-12-05T11:35:07Z ERROR near_client::sync] State sync: state download for shard 2 timed out in 10 seconds
[2019-12-05T11:35:07Z ERROR near_client::sync] State sync: state download for shard 3 timed out in 10 seconds
```
Appeared after commit `a5ab748f3484b19112932ac63fa72917e47ee2ce` but NOT related to it.
The problem is that `NetworkRequests::StateRequest` doesn't reach requested target.

### PR DESCRIPTION
**Pull request by Kouprin**
We don't assign ourselves in State Sync.
NOT related to #1824.

### CODE REVIEW

**Comment by Kouprin** [comment]
Also related to #1837.

File: chain/client/src/sync.rs [hunk 1]
```diff
@@ -629,7 +637,11 @@ impl StateSync {
                 shard_id,
                 false,
             ) {
-                Some(AccountOrPeerIdOrHash::AccountId(validator_stake.account_id.clone()))
+                if me.as_ref().map(|me| me != &validator_stake.account_id).unwrap_or(false) {
```
Comment (bowenwang1996) [inline chain/client/src/sync.rs]: Why `unwrap_or(false)`? If you are not a validator you would not request state from a validator?

Comment (Kouprin) [reply chain/client/src/sync.rs]: It makes sense.
Should it be `if me.as_ref().map(|me| me != &validator_stake.account_id).unwrap_or(true)`?

Comment (bowenwang1996) [reply chain/client/src/sync.rs]: Yes.

**Comment by Kouprin** [comment]
> Kouprin requested review from evgenykuzyakov and nearmax as code owners 1 minute ago
because of two lines test refactoring? weird

**Comment by Kouprin** [comment]
Yes. I expect it may help further in our understanding of what's going on.

**Pull request merged**